todo_app/app.py

```diff
@@ -43,7 +53,11 @@ def createnewcard ():
     @app.route('/complete_item/<id>', methods=['POST'])
     @login_required 
     def complete_item(id):
+        if current_user.role != "writer":
+            app.logger.warn(f"user {current_user.id} tried to create a card without permissions")    
```

Comment:
You should add logging of item's `id` here and elsewhere when item changes state. This will allow you to trace what happens to your items

Reply:
done